src/http_server/api/v1/metrics.c

```diff
@@ -167,6 +170,34 @@ int is_same_metric(char *s1, char *s2) {
   return 1;
 }
 
+/* derive HELP text from metricname */
+/* if help text length > 128, increase init memory for metric_helptxt */
+void metrics_help_txt(char *metric_name, flb_sds_t *metric_helptxt)
+{
+   if (strstr(metric_name, "input_bytes")) {
+        flb_sds_cat(*metric_helptxt, " Number of input bytes.\n", 24);
+    } else if (strstr(metric_name, "input_records")) {
+        flb_sds_cat(*metric_helptxt, " Number of input records.\n", 26);
+    } else if (strstr(metric_name, "output_bytes")) {
+        flb_sds_cat(*metric_helptxt, " Number of output bytes.\n", 25);
+    } else if (strstr(metric_name, "output_records")) {
+        flb_sds_cat(*metric_helptxt, " Number of output records.\n", 27);
+    } else if (strstr(metric_name, "output_errors")) {
+        flb_sds_cat(*metric_helptxt, " Number of output errors.\n", 26);
+    } else if (strstr(metric_name, "output_retries_failed")) {
+        flb_sds_cat(*metric_helptxt, " Number of output retries failed.\n", 34);
+    } else if (strstr(metric_name, "output_retries")) {
+        flb_sds_cat(*metric_helptxt, " Number of output retries.\n", 27);
+    } else if (strstr(metric_name, "output_proc_records")) {
+        flb_sds_cat(*metric_helptxt, " Number of processed output records.\n", 37);
+    } else if (strstr(metric_name, "output_proc_bytes")) {
+        flb_sds_cat(*metric_helptxt, " Number of processed output bytes.\n", 35);
+    }
+    else {
+        flb_sds_cat(*metric_helptxt, " Fluentbit metrics.\n", 20);
+    }
+}
```

Comment:
This function needs a check at the end to ensure that the `flb_sds_cat` operation succeeded. It should then probably return -1 on error, 0 on success.

Reply:
My thought was, that this memory check can be avoided in the function, because we have nearly 128 bytes for metric_helptxt, and all the prepared helptexts in the function are shorter, so there will never occur additional memory allocation for metric_helptxt.